src/netlink_pm_upstream.c

```diff
@@ -624,6 +624,58 @@ static int upstream_get_limits(struct mptcpd_pm *pm,
                                   l_free  /* destroy */) == 0;
 }
 
+static int upstream_set_flags(struct mptcpd_pm *pm,
+                              struct sockaddr const *addr,
+                              mptcpd_flags_t flags)
+{
+        /*
+          Payload (nested):
+              Local address family
+              Local address
+              Flags
+         */
+
+        // Types chosen to match MPTCP genl API.
+        uint16_t const family = mptcpd_get_addr_family(addr);
+
+        size_t const payload_size =
+                MPTCPD_NLA_ALIGN(uint16_t) // family
+                + NLA_HDRLEN + NLA_ALIGN(mptcpd_get_addr_size(addr))
```

Comment:
I was wondering why this line was different, but then saw the `sizeof()` built in to `MPTCPD_NLA_ALIGN`. I recall you mentioning some kind of `sizeof`-related bug yesterday, is that the `MPTCPD_NLA_ALIGN(addr_size)` in `upstream_add_addr()`?

Reply:
Right!  See PR #119 for my proposed fix.

Reply:
I rebased this pull request to pull in the changes in #119.  This function now uses the `MPTCPD_NLA_ALIGN_ADDR()` macro.